core/src/wsv.rs

```diff
@@ -452,15 +583,20 @@ impl WorldStateView {
         self.blocks.iter()
     }
 
-    /// Returns iterator over blockchain blocks after the block with the given `hash`
+    /// Returns a vector of blockchain blocks after the block with the given `hash`
     pub fn blocks_after_hash(
         &self,
         hash: HashOf<VersionedCommittedBlock>,
-    ) -> impl Iterator<Item = VersionedCommittedBlock> + '_ {
-        self.blocks
-            .iter()
-            .skip_while(move |block_entry| block_entry.value().header().previous_block_hash != hash)
-            .map(|block_entry| block_entry.value().clone())
+    ) -> Vec<VersionedCommittedBlock> {
+        let mut iterator = self.blocks.iter();
+        if Hash::zeroed() != hash.into() {
+            for block in &mut iterator {
+                if block.hash() == hash {
+                    break;
+                }
+            }
+        }
+        return iterator.map(|b| b.clone()).collect();
```

Comment:
This block has a lot of weird things going on. Why do we have a mutable borrow if we don't use it?

Reply:
Because *iterating* mutates the iterator. And we want to use the iterator a second time after the loop has finished.

Reply:
It's the rough equivilant of this C code.
```C
int index = 0;
for(; index < len; index++)
{
    if(check_condition(index))
    { break; }
}
memcpy(src + index, dest, len - index);